heidrun/ct_dc.rb

```diff
@@ -0,0 +1,95 @@
+
+# If more than one dc:contributor property exists then it drops the last one
+# If one or zero exist return nil
+contributor_top = lambda do |r|
+  contrib = r['dc:contributor']
+  count = r['dc:contributor'].count
+  (count > 1) ? contrib.first_value(count - 1) : contrib.last_value(0)
+end
+
+Krikri::Mapper.define(:ct_dc,
+                      :parser => Krikri::OaiDcParser,
+                      :parser_args => '//rdf:RDF/rdf:Description') do
+  provider :class => DPLA::MAP::Agent do
+    uri 'http://dp.la/api/contributor/connecticut'
+    label 'Connecticut Digital Library'
+  end
+
+  dataProvider :class => DPLA::MAP::Agent do
+    providedLabel record.field('dc:contributor').last_value
+  end
+
+  isShownAt :class => DPLA::MAP::WebResource do
+    uri record.field('dc:identifier')
+  end
+
+  preview :class => DPLA::MAP::WebResource do
+    uri record.field('dc:source')
+  end
+
+  originalRecord :class => DPLA::MAP::WebResource do
+    uri record_uri
+  end
+
+  sourceResource :class => DPLA::MAP::SourceResource do
+    collection  :class => DPLA::MAP::Collection,
+                :each => record.field('xmlns:header', 'xmlns:setSpec'),
+                :as => :coll do
+      title coll
+    end
+
+    contributor :class => DPLA::MAP::Agent,
+                :each => record.map( &contributor_top ).flatten,
```

Comment:
I think simply `record.field('dc:contributor')[0..-2]` will work here, returning the desired values as a regular Ruby `Array`.

We should keep an eye out, though `#[]` access on `ValueArray`s doesn't chain, so if we needed to do, e.g. `record.field('dc:contributor')[0..-2].match_attribute(:blah)` or similar, we will have painted ourselves into a corner. 

Still, this looks like the best solution that's readily available.